pubsub/google/cloud/pubsub_v1/subscriber/histogram.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2017, Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division
+
+
+class Histogram(object):
+    """Representation of a single histogram.
+
+    The purpose of this class is to store actual ack timing information
+    in order to predict how long to renew leases.
+
+    The default implementation uses the 99th percentile of previous ack
+    times to implicitly lease messages; however, custom
+    :class:`~.pubsub_v1.subscriber.consumer.base.BaseConsumer` subclasses
+    are free to use a different formula.
+
+    The precision of data stored is to the nearest integer. Additionally,
+    values outside the range of ``10 <= x <= 600`` are stored as ``10`` or
```

Comment:
It could? It seems unlikely to change, but the class could be more generally useful if we need the same concept elsewhere.